File: backend/services/analytics_service.py
"""
Analytics Service - Analytics operations.
"""

from __future__ import annotations

import logging
from typing import Any, Dict, List, Optional

logger = logging.getLogger(__name__)


class AnalyticsService:
    """
    Service for analytics operations.
    """

    def __init__(self):
        self._metrics: List[Dict[str, Any]] = []

    async def record(
        self,
        event_type: str,
        data: Dict[str, Any],
        user_id: Optional[int] = None,
    ) -> None:
        """Record an analytics event."""
        self._metrics.append({
            "event_type": event_type,
            "data": data,
            "user_id": user_id,
        })

    async def get_metrics(
        self,
        event_type: Optional[str] = None,
        user_id: Optional[int] = None,
        limit: int = 100,
    ) -> List[Dict[str, Any]]:
        """Get metrics."""
        metrics = self._metrics

        if event_type:
            metrics = [m for m in metrics if m.get("event_type") == event_type]
        if user_id:
            metrics = [m for m in metrics if m.get("user_id") == user_id]

        return metrics[-limit:]

    async def get_stats(self, user_id: Optional[int] = None) -> Dict[str, Any]:
        """Get analytics statistics."""
        metrics = self._metrics
        if user_id:
            metrics = [m for m in metrics if m.get("user_id") == user_id]

        return {"total_events": len(metrics)}

File: frontend/src/services/analyticsService.js
import api from "./api";

const analyticsService = {
  async getMetrics(eventType = null, userId = null, limit = 100) {
    const response = await api.get("/api/analytics/metrics", {
      params: { event_type: eventType, user_id: userId, limit },
    });
    return response.data;
  },

  async recordEvent(eventType, data, userId = null) {
    const response = await api.post("/api/analytics/record", {
      event_type: eventType,
      data,
      user_id: userId,
    });
    return response.data;
  },

  async getStats(userId = null) {
    const response = await api.get("/api/analytics/stats", {
      params: { user_id: userId },
    });
    return response.data;
  },
};

export default analyticsService;
